src/main/java/seedu/address/model/job/Job.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.model.job;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents a Job Listing in the address book.
+ */
+public class Job {
+    private final JobName name;
+    private final JobCompany company;
+    private final JobSalary salary;
+    private final JobRequirements requirements;
+    private final JobDescription description;
+
+    /**
+     * TODO: Implement data validation
+     * Every field has no validation for now.
+     */
+    public Job(JobName name, JobCompany company, JobSalary salary, JobRequirements requirements,
+            JobDescription description) {
+        this.name = name;
+        this.company = company;
+        this.salary = salary;
+        this.requirements = requirements;
+        this.description = description;
+    }
+
+    public JobName getName() {
+        return name;
+    }
+
+    public JobCompany getCompany() {
+        return company;
+    }
+
+    public JobSalary getSalary() {
+        return salary;
+    }
+
+    public JobRequirements getRequirements() {
+        return requirements;
+    }
+
+    public JobDescription getDescription() {
+        return description;
+    }
+
+    /** Returns true if both jobs have the same name and company. */
+    public boolean isSameJob(Job otherJob) {
+        if (otherJob == this) {
+            return true;
+        }
+
+        return otherJob != null && otherJob.getName().equals(getName()) && otherJob.getCompany().equals(getCompany());
+    }
+
+    /**
+     * Returns true if both jobs have the same identity and data fields.
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
I think this is implicitly used by `UniqueJobList::equals` -> `AddressBook::equals` -> `ModelManager::equals`

But I am not sure what is `ModelManager::equals` is for since I cant find its usage outside of Test